src/Models/UserIdentityModel.php

```diff
@@ -25,6 +26,146 @@ class UserIdentityModel extends Model
     ];
     protected $useTimestamps = true;
 
+    public function getAccessTokenByToken(string $token): ?AccessToken
+    {
+        return $this
+            ->where('type', 'access_token')
+            ->where('secret', hash('sha256', $token))
+            ->asObject(AccessToken::class)
+            ->first();
+    }
+
+    /**
+     * @param int|string $userId
+     */
+    public function getAccessToken($userId, string $token): ?AccessToken
```

Comment:
I ascribe to the "pass large" design philosophy, where you use the largest logical parameter for method input (here: `User $user`, or Authenticatable or whatever). This lets your validation logic remain outside the current method (I.e. you know that User exists, has a certain structure) and helps static analysis know your true intent (much less likely to pass a random string than a whole User). 

Probably you are being consistent with other methods that already exist, but if we're doing a refactor it is something I would like to consider. One "CodeIgniter-specific" issue this solves is our inconsistency around Entity primary keys - sometimes `int`, sometimes autoincrement (natural), sometimes string (UUID or other).

Reply:
Since everything else is passing `$userId`, I am not comfortable making this the only place where `User`|`Authenticatable` is used.

If we change it to pass the User, I don't think it will be a problem, but please make it a different PR.

Reply:
Sounds good, I will create an issue.

Reply:
https://github.com/lonnieezell/codeigniter-shield/issues/66